android/airship-framework-proxy/src/main/java/com/urbanairship/android/framework/proxy/proxies/LiveUpdatesManagerProxy.kt

```diff
@@ -0,0 +1,176 @@
+package com.urbanairship.android.framework.proxy.proxies
+
+import com.urbanairship.json.JsonException
+import com.urbanairship.json.JsonMap
+import com.urbanairship.json.JsonSerializable
+import com.urbanairship.json.JsonValue
+import com.urbanairship.json.jsonMapOf
+import com.urbanairship.json.optionalField
+import com.urbanairship.json.requireField
+import com.urbanairship.liveupdate.LiveUpdate
+import com.urbanairship.liveupdate.LiveUpdateManager
+import com.urbanairship.util.DateUtils
+
+public class LiveUpdatesManagerProxy(private val managerProvider: () -> LiveUpdateManager) {
+
+    private val manager: LiveUpdateManager
+        get() {
+            return managerProvider()
+        }
+
+    public suspend fun list(request: LiveUpdateRequest.List): List<LiveUpdateProxy> {
+        return this.manager.getAllActiveUpdates().filter { it.type == request.type }.map { LiveUpdateProxy(it) }
+    }
+
+    public suspend fun listAll(): List<LiveUpdateProxy> {
+        return this.manager.getAllActiveUpdates().map { LiveUpdateProxy(it) }
+    }
+
+    public fun create(request: LiveUpdateRequest.Create) {
+        this.manager.start(
+            name = request.name,
+            type = request.type,
+            content = request.content,
+            timestamp = request.timestamp ?: System.currentTimeMillis(),
+            dismissTimestamp = request.dismissalTimestamp
+        )
+    }
+
+    public fun update(request: LiveUpdateRequest.Update) {
+        this.manager.update(
+            name = request.name,
+            content = request.content,
+            timestamp = request.timestamp ?: System.currentTimeMillis(),
+            dismissTimestamp = request.dismissalTimestamp
+        )
+    }
+
+    public fun end(request: LiveUpdateRequest.End) {
+        this.manager.end(
+            name = request.name,
+            content = request.content,
+            timestamp = request.timestamp ?: System.currentTimeMillis(),
+            dismissTimestamp = request.dismissalTimestamp
+        )
+    }
+
+    public suspend fun update(request: LiveUpdateRequest.List): List<LiveUpdateProxy> {
+        return this.manager.getAllActiveUpdates().filter { it.type == request.type }.map { LiveUpdateProxy(it) }
+    }
+
+    public suspend fun end(request: LiveUpdateRequest.List): List<LiveUpdateProxy> {
+        return this.manager.getAllActiveUpdates().filter { it.type == request.type }.map { LiveUpdateProxy(it) }
```

Comment:
is there a copy-paste issue here or does this functionality need to be exposed through multiple interfaces? (`public suspend fun update(request: LiveUpdateRequest.List): List<LiveUpdateProxy>` and `public suspend fun end(request: LiveUpdateRequest.List): List<LiveUpdateProxy>` are the same function internally)

Reply:
Fixed